addons/mrp/models/mrp_production.py

```diff
@@ -388,8 +397,10 @@ def open_produce_product(self):
     @api.multi
     def button_plan(self):
         """ Create work orders. And probably do stuff, like things. """
-        orders_to_plan = self.filtered(lambda order: order.routing_id and order.state == 'confirmed')
+        orders_to_plan = self.filtered(lambda order: order.bom_id.routing_id and order.state == 'confirmed')
         for order in orders_to_plan:
+            if not order.routing_id.operation_ids:
+                raise UserError(_('Either create an operation in the routing or remove the routing linked to the BoM!'))
```

Comment:
the plan button should not appear if there are no workcenter operations on the route, so this should not be necessary